libvast/src/experimental_table_slice.cpp

```diff
@@ -297,8 +307,9 @@ auto count_at = [](const auto& arr, int64_t row) {
   return static_cast<count>(arr.Value(row));
 };
 
-auto enumeration_at = [](const auto& arr, int64_t row) {
-  return static_cast<enumeration>(arr.Value(row));
+auto enumeration_at = [](const arrow::DictionaryArray& arr, int64_t row) {
+  const auto& b = reinterpret_cast<const arrow::Int8Array&>(*arr.indices());
```

Comment:
Does `Int8Array` suffice here, because we use `uint8_t` on the VAST side of things?

Reply:
Arrow requires signed integer types for the enum indices, so I switched the data type to `int16` to make enough room for the value range used in VAST enums.